### PR TITLE
Use puppet-kmod to manage kernel_modules

### LINKED ISSUE #8: [improvement] Use puppet-kmod module to handle Kernel modules
## What are you seeing
Kernel modules are currently handled as files

## What behaviour did you expect instead
Use https://github.com/voxpupuli/puppet-kmod to do it

### PR DESCRIPTION
**Pull request by SimonHoenscheid**
Fixes #8

### CODE REVIEW

File: manifests/node/kubelet.pp [hunk 1]
```diff
@@ -135,21 +136,13 @@
...
-
-    'modprobe br_netfilter':
-      unless => 'lsmod | grep overlay';
+  if $manage_kernel_modules {
+    kmod::load { 'overlay':
+      ensure  => $ensure,
+    }
+    kmod::load { 'br_netfilter':
+      ensure  => $ensure,
+    }
```
Comment (ananace) [inline manifests/node/kubelet.pp:145]: Just to keep things consistent, I'l prefer to see this in the form of;

```puppet
kmod::load {
  default:
    ensure => $ensure;

  'overlay':;
  'br_netfilter':;
}
```

Comment (kenyon) [inline manifests/node/kubelet.pp:145]: Can also be done with iteration like this:

```suggestion
    [
      'br_netfilter',
      'overlay',
    ].each |String[1] $mod| {
      kmod::load { $mod:
        ensure => $ensure,
      }
    }
```

Comment (kenyon) [reply manifests/node/kubelet.pp:145]: Can also be done with iteration like this:

```suggestion
    [
      'br_netfilter',
      'overlay',
    ].each |String[1] $mod| {
      kmod::load { $mod:
        ensure => $ensure,
      }
    }
```

Comment (ananace) [reply manifests/node/kubelet.pp:145]: Certainly possible, but I was more commenting on keeping things consistent compared to how it's done everywhere else in the module.

Comment (bastelfreak) [reply manifests/node/kubelet.pp:145]: why so complicated? We can write it even shorter:
```puppet
kmod::load { ['br_netfilter', 'overlay']:
  ensure => $ensure,
}
```

**Review by bastelfreak — COMMENTED** [review]
please also add a unit test and add the module to .fixtures.yml

File: manifests/node/kubelet.pp (hunk 1 above)
Comment (SimonHoenscheid) [reply manifests/node/kubelet.pp:145]: I like @bastelfreak s solution. But I will go with @ananace proposal. So it stays consistent. I can make a separate PR to change all resources later.

Comment (kenyon) [reply manifests/node/kubelet.pp:145]: @bastelfreak that style is "bad" according to the style guide ;) https://www.puppet.com/docs/puppet/7/style_guide.html#multiple-resources

**Review by ananace — APPROVED** [review]

**Pull request merged**